packages/aws-cdk/lib/api/cloudformation-deployments.ts

```diff
@@ -366,6 +360,115 @@ export class CloudFormationDeployments {
     return stack.exists;
   }
 
+  private async readCurrentNestedTemplate(stackArtifact: cxapi.CloudFormationStackArtifact, nestedStackName: string): Promise<Template> {
+    return this.readCurrentStackTemplate(stackArtifact, nestedStackName);
+  }
+
+  private async replaceNestedStacksInRootStack(rootStackArtifact: cxapi.CloudFormationStackArtifact, deployedTemplate: any) {
+    if (!this.stackHasNestedStacks(rootStackArtifact)) {
+      return;
+    }
+
+    const sdkWithRole = await prepareSdkWithLookupRoleFor(this.sdkProvider, rootStackArtifact);
```

Comment:
I think the flow is weird here.

This method (`replaceNestedStacksInRootStack()`) is called from `readCurrentTemplateWithNestedStacks()`, which first calls `readCurrentStackTemplate()`, which already does the dance with `prepareSdkWithLookupRoleFor()`.

Then why are we calling `prepareSdkWithLookupRoleFor()` again here? That doesn't seem correct.

Perhaps you need to return the `ISDK` also from `readCurrentStackTemplate()`?

If you do that, pretty sure you'll be able to get rid of the `stackHasNestedStacks()` helper too (I don't think it will be needed when you'll have the `ISDK` already).